src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
@DanVanAtta Pretty sure if you actually complete your code and format it, its going to be around 10-12 lines. If it took you 10 mins to understand 13 lines of code then I don't really know what to tell you. I could write the whole thing on 1 long line and that doesn't make it better...

I refactored it to have only 1 return and break instead. Also combined the conditionals.

Well no matter what you do there is going to be more complexity as we are adding more options. However you code it you still will need to consider the new property. Look at your sample and how much more complex it got once you had to add it in.

I really think this is ready to merge now.